### PR TITLE
Complete the cookbook on category override

### PR DESCRIPTION
**Pull request by damien-carcel**
Add a new section to explain how to add a translatable property

### CODE REVIEW

File: cookbook/category/add-properties-to-category.rst [hunk 1]
```diff
@@ -7,8 +7,12 @@ The Akeneo PIM allows the classification of products inside a customizable categ
 
     To implement this task you have to be comfortable with Symfony bundle overriding and Symfony services. This cookbook can be used to override other entities such as an attribute.
 
-Add Properties to your own Category
------------------------------------
+Add a Non Translatable Properties to your own Category
```
Comment (rybus) [inline cookbook/category/add-properties-to-category.rst]: Add a [...] property
Or
Add Non Translatable properties

File: cookbook/category/add-properties-to-category.rst [hunk 2]
```diff
@@ -109,3 +113,120 @@ For the form validation you will have to add a new validation file:
     :language: yaml
     :prepend: # /src/Acme/Bundle/CatalogBundle/Resources/config/validation/category.yml
     :linenos:
+
+Add a Translatable Properties to your own Category
```
Comment (rybus) [inline cookbook/category/add-properties-to-category.rst]: Same notice about "a property"/"properties"

File: cookbook/category/add-properties-to-category.rst [hunk 3]
```diff
@@ -109,3 +113,120 @@ For the form validation you will have to add a new validation file:
...
+--------------------------------------------------
+
+Extend the category entity and its translation entity
+*****************************************************
+
+The first step is to create a class that extends PIM ``CategoryTranslation`` class.
```
Comment (rybus) [inline cookbook/category/add-properties-to-category.rst]: that extends the PIM category translation one

File: cookbook/category/add-properties-to-category.rst [hunk 4]
```diff
@@ -109,3 +113,120 @@ For the form validation you will have to add a new validation file:
...
+You need to update your category entity parameter used in ``entities.yml`` file (if you are creating a new bundle, double check that this file is inside the extension):
+
+.. literalinclude:: ../../src/Acme/Bundle/CatalogBundle/Resources/config/entities.yml
+   :language: yaml
+   :prepend: # /src/Acme/Bundle/CatalogBundle/Resources/config/entities.yml
+   :lines: 1,3,4
+   :linenos:
+
+.. note::
+   You don't have to add a lot of code to the doctrine configuration to resolve target entities.
```
Comment (rybus) [inline cookbook/category/add-properties-to-category.rst]: Doctrine

File: cookbook/category/add-properties-to-category.rst (hunk 3 above)
Comment (damien-carcel) [inline cookbook/category/add-properties-to-category.rst]: No, here it is the name of the class itself I want.

File: cookbook/category/add-properties-to-category.rst [hunk 5]
```diff
@@ -40,7 +44,7 @@ You also need to configure the mapping override in your application configuratio
                 override: Acme\Bundle\CatalogBundle\Entity\Category
 
 Define the Category Class
--------------------------
+*************************
 
 You need to update your category entity parameter used in ``entities.yml`` file (if you are creating a new bundle, double check that this file is inside the extension):
```
Comment (a2xchip) [inline cookbook/category/add-properties-to-category.rst]: May be wee can show it like

```php
// AcmeBundleExtension.php

// code
        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
        $loader->load('entities.yml');

// more code
```

File: cookbook/category/add-properties-to-category.rst [hunk 6]
```diff
@@ -109,3 +113,120 @@ For the form validation you will have to add a new validation file:
...
+                original: Pim\Bundle\CatalogBundle\Entity\Category
+                override: Acme\Bundle\CatalogBundle\Entity\Category
+            -
+                original: Pim\Bundle\CatalogBundle\Entity\CategoryTranslation
+                override: Acme\Bundle\CatalogBundle\Entity\CategoryTranslation
+
+Define the Category Class
+*************************
+
+You need to update your category entity parameter used in ``entities.yml`` file (if you are creating a new bundle, double check that this file is inside the extension):
```
Comment (a2xchip) [inline cookbook/category/add-properties-to-category.rst]: Same here

File: cookbook/category/add-properties-to-category.rst [hunk 7]
```diff
@@ -40,7 +44,7 @@ You also need to configure the mapping override in your application configuratio
                 override: Acme\Bundle\CatalogBundle\Entity\Category
 
 Define the Category Class
--------------------------
+*************************
 
 You need to update your category entity parameter used in ``entities.yml`` file (if you are creating a new bundle, double check that this file is inside the extension):
```
Comment (a2xchip) [inline cookbook/category/add-properties-to-category.rst]: May be wee can show it like

```php
// AcmeBundleExtension.php

// code
        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
        $loader->load('entities.yml');

// more code
```

or make it bold

**double check that file ```entities.yml``` is inside the extension**

File: src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        
+        $builder->add(
+            'description',
+            'pim_translatable_field',
+            [
+                'field'             => 'description',
+                'translation_class' => $this->translationdataClass,
```
Comment (a2xchip) [inline src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php]: naming typo? ```$translationDataClass``` instead?

Comment (a2xchip) [inline src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php]: It is naming issue for PIM itself ))) https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/EnrichBundle/Form/Type/CategoryType.php#L38

Comment (damien-carcel) [inline src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php]: Nice catch on this one !
I will propose a fix on master and do a PR on the doc (master) too.

File: cookbook/category/add-properties-to-category.rst (hunk 7 above)
Comment (damien-carcel) [inline cookbook/category/add-properties-to-category.rst]: image

File: src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php (hunk 8 above)
Comment (pierallard) [inline src/Acme/Bundle/EnrichBundle/Form/Type/TranslatableCategoryType.php]: @a2xchip the :eyes: of the :tiger: , thx ;)

**Comment by juliensnz** [comment]
GTM

**Pull request merged**